src/Config/Converters/AKVRetryPolicyOptionsConverterFactory.cs

```diff
@@ -0,0 +1,208 @@
+    internal AKVRetryPolicyOptionsConverterFactory(bool replaceEnvVar)
+    {
+        _replaceEnvVar = replaceEnvVar;
+    }
+
+    private class AKVRetryPolicyOptionsConverter : JsonConverter<AKVRetryPolicyOptions>
+    {
+        // Determines whether to replace environment variable with its
+        // value or not while deserializing.
+        private bool _replaceEnvVar;
+
+        /// <param name="replaceEnvVar">Whether to replace environment variable with its
+        /// value or not while deserializing.</param>
+        public AKVRetryPolicyOptionsConverter(bool replaceEnvVar)
+        {
+            _replaceEnvVar = replaceEnvVar;
+        }
+
+        /// <summary>
+        /// Defines how DAB reads AKV Retry Policy options and defines which values are
+        /// used to instantiate those options.
+        /// </summary>
+        /// <exception cref="JsonException">Thrown when improperly formatted cache options are provided.</exception>
+        public override AKVRetryPolicyOptions? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType is JsonTokenType.StartObject)
+            {
+                AKVRetryPolicyMode? mode = null;
+                int? maxCount = null;
+                int? delaySeconds = null;
+                int? maxDelaySeconds = null;
+                int? networkTimeoutSeconds = null;
+
+                while (reader.Read())
+                {
+                    if (reader.TokenType is JsonTokenType.EndObject)
+                    {
+                        return new AKVRetryPolicyOptions(mode, maxCount, delaySeconds, maxDelaySeconds, networkTimeoutSeconds);
+                    }
+
+                    string? property = reader.GetString();
+                    reader.Read();
+
+                    switch (property)
+                    {
+                        case "mode":
+                            if (reader.TokenType is JsonTokenType.Null)
+                            {
+                                mode = null;
+                            }
+                            else
+                            {
+                                mode = EnumExtensions.Deserialize<AKVRetryPolicyMode>(reader.DeserializeString(_replaceEnvVar)!);
+                            }
+
+                            break;
+                        case "max-count":
+                            if (reader.TokenType is JsonTokenType.Null)
+                            {
+                                maxCount = null;
+                            }
+                            else
+                            {
+                                int parseMaxCount = reader.GetInt32();
+                                if (parseMaxCount < 0)
+                                {
+                                    throw new JsonException($"Invalid value for max-count: {parseMaxCount}. Value must be positive.");
```

Comment:
Technically, 0 is neither positive nor negative. The message should say "Value must not be negative."